solver/src/baseline_solver.rs

```diff
@@ -43,6 +43,10 @@ impl BaselineSolvable for AmmOrder {
     fn get_spot_price(&self, base_token: H160, quote_token: H160) -> Option<BigRational> {
         amm_to_pool(self).get_spot_price(base_token, quote_token)
     }
+
+    fn gas_cost(&self) -> usize {
+        amm_to_pool(self).gas_cost()
```

Comment:
interesting I hadn't see `amm_to_pool` before. Now that I see it and realize that `AmmOrder` and `Pool` share the same attributes, couldn't we do something cute that would make them both essentially the same... I mean we should be able to just say `amm_order.gas_cost()` by making gas cost a trait method and implementing it for Pool and AmmOrder

Reply:
AmmOrder is of type `Liquidity` which at the moment is only exposed in the driver crate and includes logic on how to actually encode a swap into calldata. `BaselineSolvable` (or `Pool` in this example) is the type in the `shared` crate that is used for both the API as well as the driver's BaselineSolver.

Right now the API doesn't have to know about how a liquidity source is encoded as an `Interaction`. However, this may change in the future (if we e.g. want to simulate the Baseline execution to see if tokens can be supported) in which case the refactoring you suggest makes a lot of sense.